packages/@ember/-internals/runtime/lib/system/object.js

```diff
@@ -63,7 +63,7 @@ if (DEBUG) {
 
     [ASSERT_INIT_WAS_CALLED]() {
       assert(
-        `You must call \`this._super(...arguments);\` when overriding \`init\` on a framework object. Please update ${this} to call \`this._super(...arguments);\` from \`init\`.`,
+        `You must call \`super.init(...arguments);\` when overriding \`init\` on a framework object. Please update ${this} to call \`super.init(...arguments);\` from \`init\`.`,
```

Comment:
`super.init` is only valid when extending via native classes, and does not work properly for classes created via `Ember.Object.extend()`